.github/workflows/release.yml

```diff
@@ -0,0 +1,48 @@
+name: release
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
+
+jobs:
+  test:
+    uses: ./.github/workflows/test.yml
+  packages:
+    uses: ./.github/workflows/packages.yml
+  release:
+    needs:
+      - test
+      - packages
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: hashicorp/vault-action@v2.4.2
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          secrets: |
+            secret/apm-team/ci/apm-agent-python-pypi-prod user | TWINE_USER ;
+            secret/apm-team/ci/apm-agent-python-pypi-prod password | TWINE_PASSWORD
+      - uses: actions/download-artifact@v3
+        with:
+          name: packages
+          path: dist
```

Comment:
Question regarding this step:
- Will this step ensure the package to be downloaded is the one produced in the `.github/workflows/package.yml`? 

Just wondering if two parallel workflows run (one for the merge commit in main and another one for the git tag vx.x.x), won't be any overlap between them?

Reply:
Alright, I did some testing on a private playground repo:

Given a workflow with `push` trigger that uploads some artifacts and given a release workflow that tries to download the same artifact name. Then the following error shows:

![image](https://user-images.githubusercontent.com/16325797/216970325-bf4e1209-0c86-4798-bcb8-7937f83d32b9.png)

Downloading the artifact only worked if the workflows are "connected" like we are doing it here.

Reply:
Thanks for verifying this! 👍